From 4407b32a97579854b1dd199e016e050b347dd0fd Mon Sep 17 00:00:00 2001
From: Joey <joey@lineageos.org>
Date: Mon, 3 Jun 2019 14:51:00 +0200
Subject: [PATCH 28/40] SystemUI: sync dark mode with AOSP night mode too

LiveDisplay night mode may not be available on all devices

Change-Id: I6882bc871881e3a2385b698831aa80bd70b830fd
Signed-off-by: Joey <joey@lineageos.org>
---
 .../systemui/statusbar/phone/StatusBar.java   | 23 ++++++++++++++++++-
 1 file changed, 22 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index 88ede7088fa..93464a01e18 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -135,6 +135,7 @@ import android.widget.ImageView;
 import android.widget.TextView;
 
 import com.android.internal.annotations.VisibleForTesting;
+import com.android.internal.app.ColorDisplayController;
 import com.android.internal.colorextraction.ColorExtractor;
 import com.android.internal.logging.MetricsLogger;
 import com.android.internal.logging.nano.MetricsProto.MetricsEvent;
@@ -569,6 +570,8 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
     private boolean mKeyguardShowingMedia;
     private boolean mShowMediaMetadata;
 
+    private ColorDisplayController mColorDisplayController;
+
     private BroadcastReceiver mWallpaperChangedReceiver = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
@@ -1075,6 +1078,9 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
 
         mVisualizerView = (VisualizerView) mStatusBarWindow.findViewById(R.id.visualizerview);
 
+        mColorDisplayController = new ColorDisplayController(mContext,
+                ActivityManager.getCurrentUser());
+
         // Other icons
         mVolumeComponent = getComponent(VolumeComponent.class);
 
@@ -2225,6 +2231,21 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
         return false;
     }
 
+    private boolean isAospNightModeOn() {
+        // SystemUI is initialized before ColorDisplayService, so the service may not
+        // be ready when this is called the first time
+        if (!mColorDisplayController.isAvailable(mContext)) {
+            return false;
+        }
+        try {
+            return mColorDisplayController.isActivated();
+        } catch (NullPointerException e) {
+            Log.w(TAG, e.getMessage());
+        }
+        return false;
+    }
+
+
     private String getDarkOverlay() {
         return LineageSettings.System.getString(mContext.getContentResolver(),
                 LineageSettings.System.BERRY_DARK_OVERLAY,
@@ -4154,7 +4175,7 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
 
         switch (globalStyleSetting) {
             case 1:
-                useDarkTheme = isLiveDisplayNightModeOn();
+                useDarkTheme = isLiveDisplayNightModeOn() || isAospNightModeOn();
                 break;
             case 2:
                 useDarkTheme = false;
-- 
2.17.1

